recipes/core/busybox.yaml

```diff
@@ -31,6 +31,20 @@ buildScript: |
     makeParallel
     make CONFIG_PREFIX=${BOB_CWD}/install install
 
-packageScript: |
-    cp -a $1/install/* .
+    mkdir -p install/.bob
+
+    # busybox must be setuid
+    cat >install/.bob/busybox.device-table <<EOF
+    /usr/bin/busybox f 4755 0  0 - - - - -
+    EOF
 
+    # add configured shells
+    declare -a BB_SHELLS="/bin/sh /usr/bin/sh"
+    grep -qsE '^CONFIG_SHELL_ASH=y$' build/.config && BB_SHELLS+=" /bin/ash /usr/bin/ash"
+    grep -qsE '^CONFIG_SHELL_HUSH=y$' build/.config && BB_SHELLS+=" /bin/hush /usr/bin/hush"
+    cat >install/.bob/busybox.shell-table <<EOF
+    ${BB_SHELLS// /$'\n'}
+    EOF
```

Comment:
Even though `BB_SHELLS` is declared as array, it is not used as such. Why not just print to the file directly?

```sh
(
    printf '/bin/sh\n/usr/bin/sh\n'
    grep -qE '^CONFIG_SHELL_ASH=y$' build/.config && printf '/bin/ash\n/usr/bin/ash\n'
    grep -qE '^CONFIG_SHELL_HUSH=y$' build/.config && printf '/bin/hush\n/usr/bin/hush\n'
)  > install/.bob/busybox.shell-table
```

BTW, why the `-s` switch? Wouldn't a missing `build/.config` be an error that should be visible?

Reply:
done